### DISCUSSION TITLE
Callback for provisioning

### DISCUSSION THREAD

**Discussion by lucian-torje-siemens** [eclipse-edc/Connector#2880] [General]
Hello,
Is there a way to register an endpoint that does provisioning and deprovisioning during data transfer?

Benefit: It would allow us to extend EDC without adding our own code.

**Comment by reisman234** [comment]
Hi,
I created for a while ago this short description to use the http-provisioning extensions, maybe it helps.
[HTTP-Provisioning README](https://github.com/reisman234/DataSpaceConnector/commit/c1ac8fb23ce9686c54585872ee0ffcb8a0ef1a21)

**Reply by lucian-torje-siemens** [reply]
Thank you

**Reply by lucian-torje-siemens** [reply]
Hello @reisman234 ,
Did your code made it into EDC? 
I can only find referenced to HttpProvision in the E2E tests:

![image](https://github.com/eclipse-edc/Connector/assets/102717569/3fe454ac-cd9a-4710-9c9c-52a98a7f7b7b)

If it did, how does it handle going back to main EDC flow ? Is it similar to HttpProxy meaning that I need to explicitly call the dataplane myself?

**Reply by xdxxxdx** [reply]
Hello @reisman234 , Have you tried to deprovision the asset you provisioned? Thanks
